webpack.config.js

```diff
@@ -0,0 +1,179 @@
+    watchOptions: {
+        ignored: /node_modules/
+    },
+    output: {
+        path: `${__dirname}/dist/BootstrapTooltip`,
+        filename: `${widgetDir}/[name].js`,
+        libraryTarget: "amd"
+    },
+    optimization: {
+        minimizer: [
+            new UglifyJsPlugin({
+                test: /\.js(\?.*)?$/i
+            })
+        ]
+    },
+    module: {
+        rules: [
+            {
+                test: /\.js$/,
+                exclude: /node_modules/,
+                use: {
+                    loader: "babel-loader",
+                    options: {
+                        presets: ["@babel/preset-env"],
+                        plugins: ["@babel/plugin-proposal-object-rest-spread"]
+                    }
+                }
+            },
+            {
+                test: /\.(sa|sc|c)ss$/,
+                use: [
+                    MiniCssExtractPlugin.loader,
+                    "css-loader",
+                    {
+                        loader: "postcss-loader",
+                        options: {
+                            plugins: [
+                                require("autoprefixer"),
+                                require("postcss-clean"),
+                                require("cssnano")({
+                                    preset: [
+                                        "default",
+                                        {
+                                            discardComments: {
+                                                removeAll: true
+                                            }
+                                        }
+                                    ]
+                                })
+                            ]
+                        }
+                    },
+                    "sass-loader"
+                ]
+            },
+            {
+                test: /\.(gif|png|jpe?g|svg)$/i,
+                use: [
+                    {
+                        loader: "file-loader",
+                        options: {
+                            name: `[name].[ext]`,
+                            outputPath: `${widgetUIDir}/images`
+                        }
+                    }
+                ]
+            },
+            {
+                test: /\.html$/i,
+                use: "raw-loader"
+            }
+        ]
+    },
+    externals: [/mx|mxui|mendix|dijit|dojo|require/],
+    plugins: initPlugins()
+};
+
+function initPlugins() {
+    const plugins = [
+        new CleanWebpackPlugin(),
+        new MiniCssExtractPlugin({
+            filename: `${widgetUIDir}/BootstrapTooltip.css`
+        }),
+        new XMLPlugin({
+            files: widgetXMLFiles
+        }),
+        new webpack.ProvidePlugin({
+            $: "jquery",
+            jQuery: "jquery"
+        })
+    ];
+
+    initFileManagerPlugin(plugins);
+
+    return plugins;
+}
+
+function initFileManagerPlugin(plugins) {
+    const mxTestProjectVersions = [6, 7, 8];

```

Comment:
one way of doing, it tough for the mono repo, we list the location for the test project in the package.json